### ISSUE TITLE
Connection between NC and DS(Docker) Bad Request or Timeout Failure

### ISSUE THREAD

**Issue by Rick26L** [ONLYOFFICE/onlyoffice-owncloud#194]
Hello, 

I have problems to get the connection between NC and DS running. 

### Steps to reproduce
1. Install Nextcloud with Tech and Me Script 
2. Install Onlyoffice using the onlyoffice.sh script 
3. Try to safe the Setting in the Onlyoffice App

### Expected behaviour
The Connection between NC and Onlyoffice should be established and you are able to edit and create documents with onlyoffice.

### Actual behaviour
A Minute after safing the Settings you get the Error "Bad Request oder Timeout Fehlermeldung" 
It is not possible to open/edit documents with onlyoffice. 
The Log says: 
"file_get_contents(https://...:443/ConvertService.ashx): failed to open stream: HTTP request failed! HTTP/1.1 504 Gateway Time-out
 at /var/www/nextcloud/apps/onlyoffice/lib/documentservice.php#351"

## Server configuration detail

**Operating system:** Linux 4.15.0-20-generic #21-Ubuntu SMP Tue Apr 24 06:16:15 UTC 2018 x86_64

**Webserver:** Apache/2.4.29 (Ubuntu) (apache2handler)

**Database:** pgsql PostgreSQL 10.4 (Ubuntu 10.4-0ubuntu0.18.04) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 7.3.0-16ubuntu3) 7.3.0, 64-bit

**PHP version:** 7.2.5-0ubuntu0.18.04.1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, sodium, session, standard, apache2handler, PDO, xml, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, igbinary, imagick, imap, intl, json, ldap, exif, pdo_pgsql, pgsql, Phar, posix, readline, redis, shmop, SimpleXML, smbclient, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, libsmbclient, Zend OPcache

**Nextcloud version:** 13.0.3 - 13.0.3.2

**Updated from an older Nextcloud/ownCloud or fresh install:**  Fresh install

**Where did you install Nextcloud from:** Tech and Me 

<details><summary>Signing status</summary>

Array
</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - activity: 2.6.1
 - caniupdate: 0.1.2
 - comments: 1.3.0
 - dav: 1.4.7
 - federatedfilesharing: 1.3.1
 - federation: 1.3.0
 - files: 1.8.0
 - files_external: 1.4.1
 - files_rightclick: 0.8.4
 - files_sharing: 1.5.0
 - files_texteditor: 2.5.1
 - files_trashbin: 1.3.0
 - files_versions: 1.6.0
 - firstrunwizard: 2.2.1
 - issuetemplate: 0.3.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.1.0
 - nextcloud_announcements: 1.2.0
 - notifications: 2.1.2
 - oauth2: 1.1.1
 - onlyoffice: 1.3.0
 - password_policy: 1.3.0
 - previewgenerator: 1.0.9
 - provisioning_api: 1.3.0
 - serverinfo: 1.3.0
 - sharebymail: 1.3.0
 - survey_client: 1.1.0
 - systemtags: 1.3.0
 - theming: 1.4.5
 - twofactor_backupcodes: 1.2.3
 - updatenotification: 1.3.0
 - user_ldap: 1.3.1
 - workflowengine: 1.3.0
Disabled:
 - admin_audit
 - calendar
 - contacts
 - encryption
 - files_fulltextsearch
 - files_pdfviewer
 - files_videoplayer
 - fulltextsearch
 - fulltextsearch_elasticsearch
 - gallery
 - passman
 - user_external

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "localhost",
        "192.168.16.15",
        "cloud...",
        "office..."
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "overwrite.cli.url": "https:\/\/cloud...\/",
    "dbtype": "pgsql",
    "version": "13.0.3.2",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "log_rotate_size": "10485760",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "filelocking.enabled": true,
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 0,
        "timeout": 0.5,
        "dbindex": 0,
        "password": "***REMOVED SENSITIVE VALUE***"
    },
    "htaccess.RewriteBase": "\/",
    "loglevel": "2",
    "log_type": "file",
    "logfile": "\/mnt\/ncdata\/nextcloud.log",
    "logtimezone": "Europe\/Berlin",
    "maintenance": false,
    "ldapIgnoreNamingRules": false,
    "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
    "theme": "",
    "mail_smtpauthtype": "LOGIN",
    "onlyoffice": {
        "verify_peer_off": true
    }
}
```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/

</details>

## Client configuration

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36

Thanks in advance!

**Comment by alexanderonlyoffice** [comment]
@Rick26L, it seems that one the instances is inaccessible for another. Please send us some additional information on the problem:

1. Send a screenshot of ONLYOFFICE address settings which you enter in Nextcloud;
2. Try to check the accessibility of the Document Server for Nextcloud and vice versa: 
use `wget https://documentserver_address` from the server where Nextcloud is installed and `wget https://nextcloud_address` from the server where Document Server is installed;
3. Did you enable HTTPS on both server? If yes, please specify what SSL certificates you use.

**Comment by Rick26L** [comment]
Hi  @alexanderonlyoffice , 

thanks for the feedback. 

When I tested the connection between Onlyoffice Docker and Nextcloud. I saw that the connection in this way used the wrong IP. Onlyoffice Docker wants to access the Nextcloud with the puplic IP.
I modified the hosts file and it works. 


For all with the same Problem: 
Get the Docker ID (docker ps)
Login into Docker (docker exec -it dockerID bash)
Try to access your nextcloud (wget https://nextcloud_address)
Edit the hosts file in etc 
Check Connection again.
